packages/react-reconciler/src/ReactFiberBeginWork.old.js

```diff
@@ -2239,6 +2254,14 @@ function updateDehydratedSuspenseComponent(
   // but after we've already committed once.
   warnIfHydrating();
 
+  if ((getExecutionContext() & RetryAfterError) !== NoContext) {
```

Comment:
Would checking if `root.hydrate` is false work?

Reply:
I'm...not sure. I think?

I actually haven't got a test for this code path yet. I need to add one. I'll revisit this change once I have one.

Reply:
No that wouldn't work because root.hydrate is no longer true after the first outer level is committed. At that point "dehydrated boundaries" are marked as still hydrating. `root.hydrate` is kind of like that the root is a dehydrated boundary.